docs/fundamentals/executor/monitoring-executor.md

```diff
@@ -0,0 +1,151 @@
+please refer to {ref}`this <monitoring-flow>` section.
+````
+
+## Defining custom metrics
+
+Sometime monitoring the `enconding` method is not enough, you need to break it up into multiple parts that you want to 
+monitor one by one.
+
+It could be useful if your encoding phase for example is composed of two tasks: image processing and
+image embedding. By using custom metrics on this two tasks you could identify potential bottleneck.
+
+Overall the ability to add custom metrics allows you to have the full flexibility on the monitoring of your Executor.
+
+### Defining custom metrics with `@monitor`
+
+Let's take an example to illustrate the custom metrics
+
+```python
+from jina import requests, Executor
+from docarray import DocumentArray
+
+
+class MyExecutor(Executor):
+    def preprocessing(self, docs: DocumentArray):
+        ...
+
+    def model_inference(self, tensor):
+        ...
+
+    @requests
+    def encode(self, docs: DocumentArray, **kwargs):
+        docs.tensors = self.preprocessing(docs)
+        docs.embedding = self.model_inference(docs.tensors)
+```
+
+This Executor encodes is composed of two sub function.
+* `preprocessing` which take raw bytes from a DocumentArray and put it into a torch tensor. 
+* `model inference` call the forward function of a deep learning model.
+
+By default, only the encode function will be monitored. 
+
+````{admonition} Using @monitor
+:class: hint
+Adding the custom monitoring on a method is as straightforward as decorating the method with `@monitor` 
+````
+
+```{code-block} python
+---
+emphasize-lines: 6, 10
+---
+from jina import requests,monitor,Executor
+from docarray import DocumentArray
+
+class MyExecutor(Executor):
+
+    @monitor()
+    def preprocessing(self, docs: DocumentArray):
+        ...
+
+    @monitor()
+    def model_inference(self, tensor):
+        ...
+
+    @requests
+    def encode(self, docs: DocumentArray, **kwargs):
+        docs.tensors = self.preprocessing(docs)
+        docs.embedding = self.model_inference(docs.tensors)
+```
+
+This will create two [prometheus summaries](https://prometheus.io/docs/concepts/metric_types/#summary)
+`jina_model_inference_seconds`and `jina_preprocessing_seconds` which will keep track of the time of execution of these
+methods.
+
+By default, the name and the documentation of the metric created by `@monitor` are auto generated based on the name
+of the function. However, you can precise it by yourself by doing :
+```python
+@monitor(
+    name='my_custom_metrics_seconds', documentation='This is my custom documentation'
+)
+def method(self):
+    ...
+``` 
+
+````{admonition} respect the prometheus naming
+:class: caution
+You should respect the prometheus naming [conventions](https://prometheus.io/docs/practices/naming/#metric-names). 
+therefore because `@monitor` create a [Summary](https://prometheus.io/docs/concepts/metric_types/#summary) under the hood
+your metrics name should finish with `seconds`
+````
```

Comment:
```suggestion
````{admonition} respect Prometheus naming
:class: caution
You should respect the [Prometheus naming conventions](https://prometheus.io/docs/practices/naming/#metric-names). 
Therefore, because `@monitor` creates a [Summary](https://prometheus.io/docs/concepts/metric_types/#summary) under the hood,
your metric's name should finish with `seconds`.
````
```